### DISCUSSION TITLE
Support for sampling

### DISCUSSION THREAD

**Discussion by Arunkumarlnr** [apache/skywalking#9226] [Q&A]
I am using skywalking springboot agent for my app(v8.10.0) I want to enable sampling for my app(only 30% of data to be sampled) I have deployed skywalking via helm in k8s. As of now I have enabled them in OAP collector in the files/conf.d/oap/trace-sampling-policy-settings.yml.. Is there a way I can enable sampling in agent itself so that only that 30% of data reaches the collector from the app. Can you please advise on the same..

**Comment by Superskyyy** [comment]
Hello there, @Arunkumarlnr. You can enable agent-side sampling using the below Java agent config, from my point of view it is not recommended and you should only consider doing so when resource is scarce, as it will have side effects on metrics analysis. For a deeper understanding you should search for sampling in issues and there are many conversations on this.
<html>
<body>


agent.sample_n_per_3_secs | Negative or zero means off, by default.SAMPLE_N_PER_3_SECS means sampling N TraceSegment in 3 seconds tops. | SW_AGENT_SAMPLE
-- | -- | --



</body>
</html>

**Comment by wu-sheng** [comment]
I think the answer is as same as https://github.com/apache/skywalking/discussions/9149.
